File: FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs
namespace FRF.Core.Response
{
    public static class ErrorCodes
    {
        // Not exists errors
        public const int ArtifactNotExists = 1;
        public const int ArtifactTypeNotExists = 2;
        public const int CategoryNotExists = 3;
        public const int ProjectNotExists = 4;
        public const int ProviderNotExists = 5;
        public const int ArtifactFromAnotherProject = 6;
        public const int InvalidStartDateForProject = 7;
        public const int ResourceNotExist = 8;
        public const int ProjectResourceNotExists = 9;

        // Relation errors
        public const int RelationAlreadyExisted = 10;
        public const int RelationNotValidDifferentBaseArtifact = 11;
        public const int RelationNotValidRepeated = 12;
        public const int RelationNotValidDifferentType = 13;
        public const int RelationNotExists = 14;
        public const int RelationCycleDetected = 15;

        // User errors
        public const int InvalidCredentials = 20;
        public const int AuthenticationServerCurrentlyUnavailable = 21;
        public const int UserNotExists = 22;

        // External errors
        public const int AmazonApiError = 30;

        //Artifacts errors
        public const int InvalidArtifactSettings = 40;

        //Resources errors
        public const int ResourceNameRepeated = 50;

        //ProjectResourcecs errors
        public const int InvalidBeginDateForProjectResource = 51;
        public const int InvalidEndDateForProjectResource = 52;
    }
}

File: FiveRockingFingers/FRF.Web/ClientApp/src/ErrorCodes.ts
// Not exists errors
export const ARTIFACT_NOT_EXISTS: number = 1;
export const ARTIFACT_TYPE_NOT_EXISTS: number = 2;
export const CATEGORY_NOT_EXISTS: number = 3;
export const PROJECT_NOT_EXISTS: number = 4;
export const PROVIDER_NOT_EXISTS: number = 5;
export const ARTIFACT_FROM_ANOTHER_PROJECT: number = 6;

// Relation errors
export const RELATION_AL_READY_EXISTED: number = 10;
export const RELATION_NOT_VALID_DIFFERENT_BASE_ARTIFACT: number = 11;
export const RELATION_NOT_VALID_REPEATED = 12;
export const RELATION_NOT_VALID_DIFFERENT_TYPE = 13;
export const RELATION_NOT_EXISTS = 14;
export const RELATION_CYCLE_DETECTED = 15;

// User errors
export const INVALID_CREDENTIALS: number = 20;
export const AUTHENTICATION_SERVER_CURRENTLY_UNAVAILABLE: number = 21;
export const USER_NOT_EXISTS: number = 22;

// External errors
export const AMAZON_API_ERROR: number = 30;

//Artifacts errors
export const INVALID_ARTIFACT_SETTINGS: number = 40;